src/libs/zmbv/zmbv.cpp

```diff
@@ -437,11 +440,12 @@ bool VideoCodec::DecompressFrame(void * framedata, int size) {
 		}
 		newframe = buf1;
 		oldframe = buf2;
-		unsigned char * writeframe = newframe + pixelsize*(MAX_VECTOR+MAX_VECTOR*pitch);	
-		for (i=0;i<height;i++) {
-			memcpy(writeframe,&work[workPos],width*pixelsize);
-			writeframe+=pitch*pixelsize;
-			workPos+=width*pixelsize;
+		unsigned char *writeframe = newframe + pixelsize * (MAX_VECTOR + MAX_VECTOR * pitch);
+		const auto bytes_per_row = static_cast<int>(width) * pixelsize;
```

Comment:
Thanks for suggestions @dreamer.

I originally went with first (before finishing the PR) and also used `bytes_per_row` to increment the `workUsed` counter, however it's only an `int` so that approach generates an overflow warning there. 

I like the second one; will give that a go!